File: lib/src/infrastructure/clients/http/responses/expense/expense_response.dart
final class ExpenseResponse {
  final int id;
  final String date;
  final String value;
  final String category;
  final String createdAt;
  final String description;

  const ExpenseResponse({
    required this.id,
    required this.date,
    required this.value,
    required this.category,
    required this.createdAt,
    required this.description,
  });

  factory ExpenseResponse.fromJson(Map<String, dynamic> json) =>
      ExpenseResponse(
        id: json['id'] as int,
        date: json['date'] as String,
        value: json['value'] as String,
        category: json['category'] as String,
        createdAt: json['created_at'] as String,
        description: json['description'] as String,
      );
}

File: app/Expense/Presentation/Http/Responses/ExpenseResponse.php
<?php

declare(strict_types=1);

namespace App\Expense\Presentation\Http\Responses;

use Illuminate\Http\Request;
use Illuminate\Http\Resources\JsonApi\JsonApiResource;

final class ExpenseResponse extends JsonApiResource
{
    public function toId(Request $request): string
    {
        return (string) $this->resource->id;
    }

    public function toType(Request $request): string
    {
        return 'expenses';
    }

    public function toAttributes(Request $request): array
    {
        return [
            'amount' => $this->resource->amount,
            'occurred_on' => $this->resource->occurredOn,
            'description' => $this->resource->description,
            'category' => $this->resource->category->value,
            'created_at' => $this->resource->createdAt?->format(format: DATE_ATOM),
        ];
    }
}
